src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.ts

```diff
@@ -65,24 +70,47 @@ export class ToolbarAddContenletComponent extends BaseComponent {
     ngOnInit(): void {
         this.contentletService.structureTypeView$.subscribe(structures => {
             this.types = structures;
-            this.recent = [];
-
-            this.types = this.types.filter(structure => {
-                    if (structure.name.startsWith('RECENT')) {
-                        this.recent.push(structure);
-                    } else {
-                        structure.types.forEach(type => {
-                            this.routingService.addPortletURL(type.name, type.action);
-                        });
+
+            this.recent = structures.filter(this.isRecentContentType);
+
+            this.types = structures.filter(structure => {
+                return !this.isRecentContentType(structure);
```

Comment:
isn't the return redundant?

Reply:
Arrow function only return when you do like:

```structures.filter(structure => !this.isRecentContentType(structure))```

But turns out that Typescript won't allow to do ```!``` like that.